crates/bevy_core_pipeline/src/taa/taa.wgsl

```diff
@@ -145,52 +133,30 @@ fn taa(@location(0) uv: vec2<f32>) -> Output {
     // Constrain past sample with 3x3 YCoCg variance clipping (reduces ghosting)
     // YCoCg: https://advances.realtimerendering.com/s2014/index.html#_HIGH-QUALITY_TEMPORAL_SUPERSAMPLING, slide 33
     // Variance clipping: https://developer.download.nvidia.com/gameworks/events/GDC2016/msalvi_temporal_supersampling.pdf
-    let s_tl = sample_view_target(uv + vec2(-texel_size.x,  texel_size.y));
-    let s_tm = sample_view_target(uv + vec2( 0.0,           texel_size.y));
-    let s_tr = sample_view_target(uv + vec2( texel_size.x,  texel_size.y));
-    let s_ml = sample_view_target(uv + vec2(-texel_size.x,  0.0));
-    let s_mm = RGB_to_YCoCg(current_color);
-    let s_mr = sample_view_target(uv + vec2( texel_size.x,  0.0));
-    let s_bl = sample_view_target(uv + vec2(-texel_size.x, -texel_size.y));
-    let s_bm = sample_view_target(uv + vec2( 0.0,          -texel_size.y));
-    let s_br = sample_view_target(uv + vec2( texel_size.x, -texel_size.y));
-    let moment_1 = s_tl + s_tm + s_tr + s_ml + s_mm + s_mr + s_bl + s_bm + s_br;
-    let moment_2 = (s_tl * s_tl) + (s_tm * s_tm) + (s_tr * s_tr) + (s_ml * s_ml) + (s_mm * s_mm) + (s_mr * s_mr) + (s_bl * s_bl) + (s_bm * s_bm) + (s_br * s_br);
     let mean = moment_1 / 9.0;
     let variance = (moment_2 / 9.0) - (mean * mean);
     let std_deviation = sqrt(max(variance, vec3(0.0)));
-    history_color = RGB_to_YCoCg(history_color);
```

Comment:
the rgb_to_ycocg is gone, but there is still a ycocg_to_rgb at the end, is that intended?

Reply:
Yes. History is stored in YCoCg. All input read from the view target needs to be converted from RGB and output stored to the view need to be converted to RGB. I think it works fine the way I have it setup, right?

Reply:
but where do you convert it to ycocg? sorry if i'm being dense, i am not great on colourspace stuff

Reply:
In sample_view_target(). The input from view target comes in as RGB, and we convert it to YCoCg, and from then on it's kept in YCoCg until we need to convert it back to write to the view target at the very end. But it goes back into the history kept as YCoCg.